File: ZLForm/Classes/ZLFormValidator.m
//
//  ZLFormValidator.m
//  ZLForm
//
//  Created by admin on 2026/5/18.
//

#import "ZLFormValidator.h"
#import "ZLFormRowDescriptor.h"

@implementation ZLFormValidator
- (instancetype)initWithMsg:(NSString *)msg validationBlock:(BOOL (^)(id value))validationBlock{
    self = [super init];
    if (self) {
        self.msg = msg;
        self.validationBlock = validationBlock;
    }
    return self;
}
- (ZLFormValidationStatus *)validate:(ZLFormRowDescriptor *)rowDescriptor{
    BOOL isValid = NO;
    if (rowDescriptor.required) {
        if (self.validationBlock) {
             isValid = self.validationBlock(rowDescriptor.value);
        }
    }else {
        isValid = YES;
    }
    ZLFormValidationStatus *status = [ZLFormValidationStatus formValidationStatusWithMsg:self.msg status:isValid rowDescriptor:rowDescriptor];
    return status;
}
@end
@implementation ZLFormValidationStatus
-(instancetype)initWithMsg:(NSString*)msg status:(BOOL)isValid rowDescriptor:(ZLFormRowDescriptor *)row {
    self = [super init];
    if (self) {
        self.msg = msg;
        self.isValid = isValid;
        self.rowDescriptor = row;
    }
    return self;
}
+(ZLFormValidationStatus *)formValidationStatusWithMsg:(NSString *)msg status:(BOOL)status rowDescriptor:(ZLFormRowDescriptor *)row {
    return [[self alloc] initWithMsg:msg status:status rowDescriptor:row];
}
@end

File: ZLForm/Classes/ZLFormValidator.swift
import UIKit

// MARK: - ZLFormValidationStatus

@objcMembers
public class ZLFormValidationStatus: NSObject {
    public var msg: String = ""
    public var isValid: Bool = false
    public weak var rowDescriptor: ZLFormRowDescriptor?
    
    public init(msg: String, status: Bool, rowDescriptor: ZLFormRowDescriptor?) {
        self.msg = msg
        self.isValid = status
        self.rowDescriptor = rowDescriptor
        super.init()
    }
    
    @objc public class func formValidationStatus(msg: String, status: Bool, rowDescriptor: ZLFormRowDescriptor) -> ZLFormValidationStatus {
        return ZLFormValidationStatus(msg: msg, status: status, rowDescriptor: rowDescriptor)
    }
}

// MARK: - ZLFormValidator

@objcMembers
public class ZLFormValidator: NSObject {
    public var msg: String = ""
    public var validationBlock: ((Any?) -> Bool)?
    
    public init(msg: String, validationBlock: @escaping (Any?) -> Bool) {
        self.msg = msg
        self.validationBlock = validationBlock
        super.init()
    }
    
    public func validate(_ rowDescriptor: ZLFormRowDescriptor) -> ZLFormValidationStatus {
        var isValid = true
        if rowDescriptor.required {
            if let block = validationBlock {
                isValid = block(rowDescriptor.value)
            }
        }
        return ZLFormValidationStatus(msg: msg, status: isValid, rowDescriptor: rowDescriptor)
    }
}
